modin/engines/ray/generic/io.py

```diff
@@ -211,25 +211,54 @@ def read_parquet(cls, path, engine, columns, **kwargs):
             columns[i : i + column_splits]
             for i in range(0, len(columns), column_splits)
         ]
-        # Each item in this list will be a list of columns of original df
-        # partitioned to smaller pieces along rows.
-        # We need to transpose the oids array to fit our schema.
-        # TODO (williamma12): This part can be parallelized even more if we
-        # separate the partitioned parquet file code path from the default one.
-        blk_partitions = np.array(
-            [
-                cls.read_parquet_remote_task._remote(
-                    args=(path, cols + partitioned_columns, num_splits, kwargs),
-                    num_return_vals=num_splits + 1,
-                )
-                if directory and cols == col_partitions[len(col_partitions) - 1]
-                else cls.read_parquet_remote_task._remote(
-                    args=(path, cols, num_splits, kwargs),
-                    num_return_vals=num_splits + 1,
-                )
-                for cols in col_partitions
+
+        # Only read row groups in parallel if parquet files are not in a partitioned within a directory
+        if not directory:
+            pf = ParquetFile(path, memory_map=False)
+            num_row_groups = pf.metadata.num_row_groups
+            blocks_with_lengths = [
+                [
+                    cls.read_parquet_remote_task._remote(
+                        args=(path, col_partitions[i], 1, j, kwargs), num_return_vals=2
+                    )
+                    for i in range(len(col_partitions))
+                ]
+                for j in range(num_row_groups)
```

Comment:
Instead of reading one row group at a time, can we read a group of row groups? Similar to how we read multiple columns at a time.